### PR TITLE
storage: allow to batch splits retrieval for multiple metrics

### PR DESCRIPTION
**Pull request by jd**
It also fixes the parallel_map call to be unique, making _get_splits a little
bit faster for unbatched drivers.

### CODE REVIEW

**Comment by chungg** [comment]
error looks legit

**Comment by jd** [comment]
Pull-request updated, HEAD is now b997ad747c89d0597500a1519f73d4a0ba53eb2c

**Comment by jd** [comment]
Rebased

**Comment by jd** [comment]
Fix pep8

File: gnocchi/storage/__init__.py [hunk 1]
```diff
@@ -365,35 +378,37 @@ def _update_metric_splits(self, metrics_keys_aggregations_splits):
...
+                    # them fully. First, fetch all those existing splits.
+                    keys_to_get[metric][aggregation].append(key)
+                    splits_to_rewrite[metric][aggregation].append(split)
+
+        existing_data = self._get_splits_and_unserialize(keys_to_get)
+
+        for metric, (keys_and_aggregations_and_splits,
+                     oldest_mutable_timestamp) in six.iteritems(
+                         metrics_keys_aggregations_splits):
+            for aggregation, existing_list in six.iteritems(
```
Comment (chungg) [inline gnocchi/storage/__init__.py]: :/ we're doing a lot of extra looping to build format. hope it doesn't affect the unbatched routes.

**Pull request merged**